client/client.go

```diff
@@ -231,53 +270,189 @@ type (
 
 	// GetWorkerBuildIdCompatibilityOptions is the input to Client.GetWorkerBuildIdCompatibility.
-	// WARNING: Worker versioning is currently experimental
+	// WARNING: Legacy worker versioning is currently experimental
 	GetWorkerBuildIdCompatibilityOptions = internal.GetWorkerBuildIdCompatibilityOptions
 
 	// WorkerBuildIDVersionSets is the response for Client.GetWorkerBuildIdCompatibility.
-	// WARNING: Worker versioning is currently experimental
+	// WARNING: Legacy worker versioning is currently experimental
 	WorkerBuildIDVersionSets = internal.WorkerBuildIDVersionSets
 
 	// BuildIDOpAddNewIDInNewDefaultSet is an operation for UpdateWorkerBuildIdCompatibilityOptions
 	// to add a new BuildID in a new default set.
-	// WARNING: Worker versioning is currently experimental
+	// WARNING: Legacy worker versioning is currently experimental
 	BuildIDOpAddNewIDInNewDefaultSet = internal.BuildIDOpAddNewIDInNewDefaultSet
 
 	// BuildIDOpAddNewCompatibleVersion is an operation for UpdateWorkerBuildIdCompatibilityOptions
 	// to add a new BuildID to an existing compatible set.
-	// WARNING: Worker versioning is currently experimental
+	// WARNING: Legacy worker versioning is currently experimental
 	BuildIDOpAddNewCompatibleVersion = internal.BuildIDOpAddNewCompatibleVersion
 
 	// BuildIDOpPromoteSet is an operation for UpdateWorkerBuildIdCompatibilityOptions to promote a
 	// set to be the default set by targeting an existing BuildID.
-	// WARNING: Worker versioning is currently experimental
+	// WARNING: Legacy worker versioning is currently experimental
 	BuildIDOpPromoteSet = internal.BuildIDOpPromoteSet
 
 	// BuildIDOpPromoteIDWithinSet is an operation for UpdateWorkerBuildIdCompatibilityOptions to
 	// promote a BuildID within a set to be the default.
-	// WARNING: Worker versioning is currently experimental
+	// WARNING: Legacy worker versioning is currently experimental
 	BuildIDOpPromoteIDWithinSet = internal.BuildIDOpPromoteIDWithinSet
 
 	// GetWorkerTaskReachabilityOptions is the input to Client.GetWorkerTaskReachability.
-	// WARNING: Worker versioning is currently experimental
+	// WARNING: Legacy worker versioning is currently experimental
 	GetWorkerTaskReachabilityOptions = internal.GetWorkerTaskReachabilityOptions
 
 	// WorkerTaskReachability is the response for Client.GetWorkerTaskReachability.
-	// WARNING: Worker versioning is currently experimental
+	// WARNING: Legacy worker versioning is currently experimental
 	WorkerTaskReachability = internal.WorkerTaskReachability
 
 	// BuildIDReachability describes the reachability of a buildID
-	// WARNING: Worker versioning is currently experimental
+	// WARNING: Legacy worker versioning is currently experimental
 	BuildIDReachability = internal.BuildIDReachability
 
 	// TaskQueueReachability Describes how the Build ID may be reachable from the task queue.
-	// WARNING: Worker versioning is currently experimental
+	// WARNING: Legacy worker versioning is currently experimental
```

Comment:
Is "Legacy worker"  versioning deprecated?

Reply:
Yes, the docs team are going to use "Legacy worker versioning" or "Legacy versioning" for the deprecated one, and plain "worker versioning"  for the current one.

Reply:
Do you want me to mark all the constants as `Deprecated:` and remove the experimental?

Reply:
Yes I think we should notify users these APIs are deprecated, do we plan to remove these APIs at some point as well? if so we should also note that in the deprecation  description.

Reply:
There is no clear plans when or how will get removed, but I will make it explicit that they are being replaced by the new versioning apis